src/twisted/web/http_headers.py

```diff
@@ -235,12 +271,14 @@ def addRawHeader(self, name, value):
 
         self.setRawHeaders(name, values)
 
-    def getRawHeaders(self, name, default=None):
+    def getRawHeaders(
+        self, name: Union[bytes, str], default: Optional[Any] = None
+    ) -> Optional[Union[List[Union[bytes, str]], Any]]:
```

Comment:
`Union[bytes, str]` ➜ `AnyStr`

Isn't `Any` under-constraining the values here?  Aren't the values also strings?

Reply:
The default is `Any`. Would a type parameter be the correct way to model this?

Reply:
In `test_getRawHeadersDefaultValue(`, https://github.com/twisted/twisted/blob/bc96c774be6e307e8e3e4d39780d37b045d0973a/src/twisted/web/test/test_http_headers.py#L115 

the default parameter of type `object()` is passed, so I think this needs to by `Any`.

Reply:
No, that's a bd reason.

I mean the test is doing it wrong.  I think it's fine that the test is testing what happens when you do it wrong, or cheating to make sure `default` is pass along as-is.  The solution is to ignore or cast the type in the test (`default = cast(bytes, object())` or something).

Don't pass on the wrong to the client, though.